From 8479708726e5ee17f4d666cf12b552d8c62eefe0 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Wed, 4 Jul 2012 12:47:40 +0800
Subject: [PATCH 2/2] LTTng: set cpu_hz before it is used.

We have to set cpu_hz before it is used in trace_clock_frequency,
otherwise the following calltrace wil be seen in precalc_stsc_interval.

WARNING: at kernel/trace/trace-clock-32-to-64.c:173 init_synthetic_tsc+0xa8/0xc0()
Modules linked in:
[<c00174c0>] (unwind_backtrace+0x0/0x104) from [<c061f40c>] (dump_stack+0x20/0x24)
[<c061f40c>] (dump_stack+0x20/0x24) from [<c003a71c>] (warn_slowpath_common+0x64/0x74)
[<c003a71c>] (warn_slowpath_common+0x64/0x74) from [<c003a758>] (warn_slowpath_null+0x2c/0x34)
[<c003a758>] (warn_slowpath_null+0x2c/0x34) from [<c0862ce4>] (init_synthetic_tsc+0xa8/0xc0)
[<c0862ce4>] (init_synthetic_tsc+0xa8/0xc0) from [<c00086a0>] (do_one_initcall+0x44/0x180)
[<c00086a0>] (do_one_initcall+0x44/0x180) from [<c084d904>] (kernel_init+0xc8/0x240)
[<c084d904>] (kernel_init+0xc8/0x240) from [<c000f83c>] (kernel_thread_exit+0x0/0x8)

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/mach-omap2/clock3xxx_data.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/clock3xxx_data.c b/arch/arm/mach-omap2/clock3xxx_data.c
index f4a626f..ba6556b 100644
--- a/arch/arm/mach-omap2/clock3xxx_data.c
+++ b/arch/arm/mach-omap2/clock3xxx_data.c
@@ -23,6 +23,7 @@
 
 #include <plat/hardware.h>
 #include <plat/clkdev_omap.h>
+#include <asm/trace-clock.h>
 
 #include "iomap.h"
 #include "clock.h"
@@ -3611,6 +3612,9 @@ int __init omap3xxx_clk_init(void)
 		(osc_sys_ck.rate / 1000000), (osc_sys_ck.rate / 100000) % 10,
 		(core_ck.rate / 1000000), (arm_fck.rate / 1000000));
 
+	/* Set cpu_hz for trace clock before it is used */
+	cpu_hz = arm_fck.rate;
+
 	/*
 	 * Only enable those clocks we will need, let the drivers
 	 * enable other clocks as necessary
-- 
1.7.7

